### PR TITLE
Update developer docs

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This PR adds a basic repo overview to the main README and mentions the `worker-name` label.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,5 +20,21 @@ This covers the existing, developer focusing, documentation:
...
 - [Resource handling](docs/resource-handling.md)
 
+# Repository layout
 
-Customer facing documentation can be found at https://github.com/kubermatic/docs which gets published at https://docs.kubermatic.io 
+```
+├── addons            # Default Kubernetes addons
+├── api 							# All the code. If you are a dev, you can initially ignore everything else
+├── CHANGELOG.md      # The changelog
+├── config            # The helmcharts we use to deploy, gets exported to https://github.com/kubermatic/kubermatic-installer
```
Comment (xrstf) [inline README.md]: Helm charts

Comment (alvaroaleman) [reply README.md]: Updated

**Pull request merged**